### ISSUE TITLE
Slow metadata query blocking query execution

### ISSUE THREAD

**Issue by premyslc** [dbeaver/dbeaver#9858]
#### System information: 
- Kubuntu 18.04
- DBeaver Version 7.2.1.202009201907

#### Connection specification:
- Oracle 11 - 12
- com.oracle.database.jdbc ojdbc8, 12.2.0.1

When i execute first statement after connect, it takes long time (93s), because there is is 
metatada query running. But it's slow only first time. Problem is, that it's not running on background and
it blocks execution of already typed sql statement where a don't need autocomplete data at this time.
An autocomplete popups shows after query is completed.

Metadata query (completes after 93s)
```
SELECT  O.*,
t.TABLE_TYPE_OWNER,t.TABLE_TYPE,t.TABLESPACE_NAME,t.PARTITIONED,t.IOT_TYPE,t.IOT_NAME,t.TEMPORARY,t.SECONDARY,t.NESTED,t.NUM_ROWS
FROM DBA_OBJECTS O
LEFT OUTER JOIN DBA_ALL_TABLES t ON (t.OWNER = O.OWNER AND t.TABLE_NAME = o.OBJECT_NAME)
WHERE O.OWNER='USERNAME' AND O.OBJECT_TYPE IN ('TABLE', 'VIEW', 'MATERIALIZED VIEW')
```

**Comment by kseniiaguzeeva** [comment]
Thank you for the report.

**Comment by LonwoLonwo** [comment]
@premyslc  We are trying to improve our query to get the table cache. Sometimes new queries work faster for someone, but unfortunately for someone slower. Can you try how long it takes for you to execute this expression?

SELECT t.OWNER, t.TABLE_NAME AS TABLE_NAME, 'TABLE' AS OBJECT_TYPE, 'VALID' AS STATUS, t.TABLE_TYPE_OWNER, t.TABLE_TYPE, t.TABLESPACE_NAME, t.PARTITIONED, t.IOT_TYPE, t.IOT_NAME, t.TEMPORARY, t.SECONDARY, t.NESTED, t.NUM_ROWS
FROM DBA_ALL_TABLES t
WHERE t.OWNER = 'KBIBLIVE'
AND NESTED = 'NO'
UNION ALL SELECT o.OWNER, o.OBJECT_NAME AS TABLE_NAME, 'VIEW' AS OBJECT_TYPE, o.STATUS, NULL, NULL, NULL, 'NO', NULL, NULL, o.TEMPORARY, o.SECONDARY, 'NO', 0
FROM DBA_OBJECTS o
WHERE o.OWNER = 'KBIBLIVE'
AND o.OBJECT_TYPE = 'VIEW'
UNION ALL SELECT o.OWNER, o.OBJECT_NAME AS TABLE_NAME, 'MATERIALIZED VIEW' AS OBJECT_TYPE, o.STATUS, NULL, NULL, NULL, 'NO', NULL, NULL, o.TEMPORARY, o.SECONDARY, 'NO', 0
FROM DBA_OBJECTS o
WHERE o.OWNER = 'KBIBLIVE'
AND o.OBJECT_TYPE = 'MATERIALIZED VIEW'

Or suggest your own version of improvement?

**Comment by premyslc** [comment]
This query seems to be fast. It is executed during 1.5s including fetch. It should be fine in combination with background cache refresh.

**Comment by LonwoLonwo** [comment]
@premyslc wow. 1.5s vs 93 second earlier? are you sure?

**Comment by premyslc** [comment]
Yes i'm. :-) But i tested query in sqlplus. It is slightly slower in DBeaver, because of formating and etc. But 5s in DBeaver looks good too.

**Comment by LonwoLonwo** [comment]
So, we added in Oracle Connection Settings in Oracle properties tab new checkbox
"Use Union for table metadata reading". The default value for this option is false. You can activate it to speed up reading the metadata.
Your feedback will be very important to us.
![2020-11-08 12_57_58-DBeaver 7 2 5 - upsert_test](https://user-images.githubusercontent.com/45152336/98462637-cf4b6c80-21c6-11eb-85dc-01deec1f079b.png)

**Comment by LonwoLonwo** [comment]
Will be implemented in 7.2.5

**Comment by kseniiaguzeeva** [comment]
verified
